source/adapters/cuda/event.cpp

```diff
@@ -248,17 +245,19 @@ urEventWait(uint32_t numEvents, const ur_event_handle_t *phEventWaitList) {
 UR_APIEXPORT ur_result_t UR_APICALL urEventRetain(ur_event_handle_t hEvent) {
   const auto RefCount = hEvent->incrementReferenceCount();
 
-  detail::ur::assertion(RefCount != 0,
-                        "Reference count overflow detected in urEventRetain.");
```

Comment:
This removes a specific error message with a generic one that does not relate to the specific error message.
If you want to keep specific errors then use setErrorMessage etc as e.g here: https://github.com/intel/llvm/pull/10626

Reply:
There are quite a few examples of this in this pr.

Reply:
> This removes a specific error message with a generic one that does not relate to the specific error message.

Agree.